backend/operations_scanner.go

```diff
@@ -147,11 +226,24 @@ loop:
 // Join waits for the OperationsScanner to gracefully shut down.
 func (s *OperationsScanner) Join() {
 	s.subscriptionWorkers.Wait()
+	s.leaderGauge.Set(0)
+}
+
+func (s *OperationsScanner) updateOperationMetrics(label string) func() {
+	startTime := time.Now()
+	s.operationsCount.WithLabelValues(label).Inc()
+	return func() {
+		s.operationsDuration.WithLabelValues(label).Observe(time.Since(startTime).Seconds())
+		s.lastOperationTimestamp.WithLabelValues(label).SetToCurrentTime()
+	}
 }
 
 // collectSubscriptions builds an internal list of Azure subscription IDs by
 // querying Cosmos DB.
 func (s *OperationsScanner) collectSubscriptions(ctx context.Context, logger *slog.Logger) {
+	done := s.updateOperationMetrics(collectSubscriptionsLabel)
+	defer done()
```

Comment:
nit: Could we compact this even further to
```
defer s.updateOperationMetrics(collectSubscriptionsLabel)()
```

Reply:
Done. The API is a bit error-prone because `defer s.updateOperationMetrics(collectSubscriptionsLabel)` compiles though it's not doing what you expect but it's probably fine if the service doesn't run new types of operation in the future.

Reply:
I'm adding a new operation now, actually, but I'll be careful to abide by this.